File: mobile-client/android/UseCase/System/Entitlement/RefreshEntitlementSnapshotUseCase.kt
package app.mobiling.client.client.usecase.system.entitlement

import app.mobiling.client.client.contract.system.entitlement.EntitlementSnapshot
import app.mobiling.client.client.data.system.entitlement.EntitlementSnapshotGateway

/**
 * Canonical use case for refreshing entitlement state.
 */
class RefreshEntitlementSnapshotUseCase(
    private val gateway: EntitlementSnapshotGateway = EntitlementSnapshotGateway(),
) {
    operator fun invoke(subjectId: String): EntitlementSnapshot = gateway.fetch(subjectId)
}

File: client/ios/UseCase/System/Entitlement/RefreshEntitlementSnapshotUseCase.swift
import Foundation

public struct RefreshEntitlementSnapshotUseCase {
    private let gateway: EntitlementSnapshotGateway

    public init(gateway: EntitlementSnapshotGateway = EntitlementSnapshotGateway()) {
        self.gateway = gateway
    }

    public func callAsFunction(subjectId: String) -> EntitlementSnapshot {
        gateway.fetch(subjectId: subjectId)
    }
}
